test/clients.go

```diff
@@ -48,56 +48,56 @@ type Clients struct {
 // and register with the Context
 type CleanupFunc func() error
 
-var contexts []*Context
+var clientsList []*Clients
```

Comment:
I'd say the word `List` is quite redundant here, isn't that an impl detail?. Using just `clients` would be just fine, IMO. Same for the other functions which include "clientsList".

Otherwise looks good.

Reply:
it is a list of "Clients", which are already plural, but sure...